cuML/src/pca/pca.h

```diff
@@ -128,23 +132,16 @@ void pcaFit(math_t *input, math_t *components, math_t *explained_var,
  * @output param mu: mean of all the features (all the columns in the data). Size n_cols * 1.
  * @output param noise_vars: variance of the noise. Size 1 * 1 (scalar).
  * @input param prms: data structure that includes all the parameters from input size to algorithm.
- * @input param cublas_handle: cublas handle
- * @input param cusolver_handle: cusolver handle
  */
 template<typename math_t>
-void pcaFitTransform(math_t *input, math_t *trans_input, math_t *components,
+void pcaFitTransform(const cumlHandle_impl& handle, math_t *input, math_t *trans_input, math_t *components,
 		math_t *explained_var, math_t *explained_var_ratio,
-		math_t *singular_vals, math_t *mu, math_t *noise_vars, paramsPCA prms,
-		cublasHandle_t cublas_handle, cusolverDnHandle_t cusolver_handle) {
-
-	pcaFit(input, components, explained_var, explained_var_ratio, singular_vals,
-			mu, noise_vars, prms, cublas_handle, cusolver_handle);
-
-	pcaTransform(input, components, trans_input, singular_vals, mu, prms,
-			cublas_handle);
-
-	signFlip(trans_input, prms.n_rows, prms.n_components, components,
-			prms.n_cols);
+		math_t *singular_vals, math_t *mu, math_t *noise_vars, paramsPCA prms) {
+    pcaFit(handle, input, components, explained_var, explained_var_ratio, singular_vals,
+           mu, noise_vars, prms);
+    pcaTransform(handle, input, components, trans_input, singular_vals, mu, prms);
+    signFlip(trans_input, prms.n_rows, prms.n_components, components,
```

Comment:
`signFlip` is neither taking a stream nor a `cumlHandle_impl` as input. IMHO all cuML/ml-prims functions which are not part of the public interface should take a stream as an argument. If they also need access to a cuBLAS handle or something similar they should take `cumlHandle_impl` additionally.
Looking at the rest of this PR I think there are more cases similar to this so please treat this as a proxy.

Reply:
That's a good catch. Thank you, Jiri. I'm sure these are coming from the `ml-prims/src/matrix` folder. Let me hunt such cases down.